analyzers/src/SonarAnalyzer.CSharp/Rules/CommentedOutCode.cs

```diff
@@ -70,57 +71,44 @@ private static void CheckTrivias(SyntaxTreeAnalysisContext context, IEnumerable<
                 else if (shouldReport
                     && trivia.IsKind(SyntaxKind.SingleLineCommentTrivia)
                     && !trivia.ToFullString().TrimStart().StartsWith("///", StringComparison.Ordinal)
-                    && IsCode(GetTriviaContent(trivia)))
+                    && IsCode(trivia.ToString().Substring(CommentMarkLength)))
                 {
                     context.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, trivia.GetLocation()));
                     shouldReport = false;
                 }
             }
         }
 
-        private static void CheckMultilineComment(SyntaxTreeAnalysisContext context, SyntaxTrivia comment)
+        private static void CheckMultilineComment(SyntaxTreeAnalysisContext context, SyntaxTrivia trivia)
         {
-            var triviaContent = GetTriviaContent(comment);
-            var triviaLines = triviaContent.Split(MetricsBase.LineTerminators, StringSplitOptions.None);
+            var triviaLines = TriviaContent().Split(MetricsBase.LineTerminators, StringSplitOptions.None);
 
             for (var triviaLineNumber = 0; triviaLineNumber < triviaLines.Length; triviaLineNumber++)
             {
                 if (IsCode(triviaLines[triviaLineNumber]))
                 {
-                    var triviaStartingLineNumber = comment.GetLocation().GetLineSpan().StartLinePosition.Line;
+                    var triviaStartingLineNumber = trivia.GetLocation().GetLineSpan().StartLinePosition.Line;
                     var lineNumber = triviaStartingLineNumber + triviaLineNumber;
                     var lineSpan = context.Tree.GetText().Lines[lineNumber].Span;
-                    var commentLineSpan = lineSpan.Intersection(comment.GetLocation().SourceSpan);
+                    var commentLineSpan = lineSpan.Intersection(trivia.GetLocation().SourceSpan);
                     var location = Location.Create(context.Tree, commentLineSpan ?? lineSpan);
                     context.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, location));
                     return;
                 }
             }
-        }
 
-        private static string GetTriviaContent(SyntaxTrivia trivia)
-        {
-            const int commentLength = 2;
-            var content = trivia.ToString();
-            if (trivia.IsKind(SyntaxKind.MultiLineCommentTrivia))
+            string TriviaContent()
```

Comment:
I suggest to use a static local functions instead of a context-capturing local functions (which behind the scenes need another class to be created) - especially for a one-liner function. The static local function is implemented as a private method by the compiler. Context-capturing needs another structure for the context. See on [sharplab](https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEBMBGAsAKHQGYACLEgdSgBcALAYQgAdqBXAJ1gHMSBvAkoLKl0AFhIBZABQBKPgKGLY1EgA8SAXhLYA3AsWDlJAJ6aSjFhwCm9APYxqV1dVl78BofoNGLbdjftHZ1lNAD41EmQSTDdFAF8CBPwCYjJMEgA5W18Obnl3IVSxSRD+Ao8jdS1dLyUHEgAvMwBlaghqKABjABlbTogAG3QAVg77KVUZWI8SWsLsADYSI1b2rt7+odGoccq5DXD1KJivJKSgA==).

I suggest to rename it to `TrimCommentMarks` and then `TrimCommentMarks(trivia)` would be easier to read then `TriviaContent()` which is a method that doesn't contain a verb (code smell).

Reply:
Using static local function would shadow the input argument and add unneeded indirection.
The verb is represented by `()` invocation, meaning `Get`.

The MSIL overhead under the hood is pretty small compared to what we do in `TokenTypeAnalyzerBase` where we create class with several collection fields for each token in source code.

Reply:
I won't insist, it's a matter of preference.

The shadowing of local input doesn't need to happen - you just use a different name.

Reply:
And I don't agree with the verb being represented by `()` .